diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index bbf30bac..b07cd4e7 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -69,6 +69,9 @@ INSTALL (TARGETS libsolv LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR} ARCHIVE DES
 
 IF (ENABLE_STATIC AND NOT DISABLE_SHARED)
 ADD_LIBRARY (libsolv_static STATIC ${libsolv_SRCS})
+IF (WIN32)
+  TARGET_COMPILE_DEFINITIONS(libsolv_static PUBLIC SOLV_STATIC_LIB)
+ENDIF (WIN32)
 SET_TARGET_PROPERTIES(libsolv_static PROPERTIES OUTPUT_NAME "solv")
 SET_TARGET_PROPERTIES(libsolv_static PROPERTIES SOVERSION ${LIBSOLV_SOVERSION})
 INSTALL (TARGETS libsolv_static LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR} ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
diff --git a/win32/regex.h b/win32/regex.h
index 06a10906..534bde74 100644
--- a/win32/regex.h
+++ b/win32/regex.h
@@ -12,7 +12,7 @@ extern "C" {
 #define CHARCLASS_NAME_MAX 14
 #define RE_DUP_MAX 255
 
-#include <BaseTsd.h>
+#include <basetsd.h>
 typedef SSIZE_T regoff_t;
 // #include <bits/alltypes.h>
 
